packages/nimble-components/src/table-column/icon/tests/table-column-icon.spec.ts

```diff
@@ -0,0 +1,467 @@
+import { html, repeat, ref } from '@microsoft/fast-element';
+import { Table, tableTag } from '../../../table';
+import { TableColumnIcon, tableColumnIconTag } from '..';
+import { waitForUpdatesAsync } from '../../../testing/async-helpers';
+import { type Fixture, fixture } from '../../../utilities/tests/fixture';
+import type { TableRecord } from '../../../table/types';
+import { TablePageObject } from '../../../table/testing/table.pageobject';
+import { wackyStrings } from '../../../utilities/tests/wacky-strings';
+import { getSpecTypeByNamedList } from '../../../utilities/tests/parameterized';
+import { mappingTextTag } from '../../../mapping/text';
+import { MappingIcon, mappingIconTag } from '../../../mapping/icon';
+import { IconXmark, iconXmarkTag } from '../../../icons/xmark';
+import { IconCheck, iconCheckTag } from '../../../icons/check';
+import type { MappingKey } from '../../../mapping/base/types';
+import { IconSeverity } from '../../../icon-base/types';
+
+interface SimpleTableRecord extends TableRecord {
+    field1?: MappingKey | undefined;
+    field2?: MappingKey | undefined;
+}
+
+interface BasicIconMapping {
+    key?: MappingKey;
+    text?: string;
+    icon: string;
+}
+
+class Model {
+    public col1!: TableColumnIcon;
+}
+interface ModelFixture<T> extends Fixture<T> {
+    model: Model;
+}
+
+describe('TableColumnIcon', () => {
+    let element: Table<SimpleTableRecord>;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TablePageObject<SimpleTableRecord>;
+    let model: Model;
+
+    // prettier-ignore
+    async function setup(mappings: BasicIconMapping[], keyType = 'string'): Promise<ModelFixture<Table<SimpleTableRecord>>> {
+        const source = new Model();
+        const result = await fixture<Table<SimpleTableRecord>>(html<Model>`
+            <${tableTag} style="width: 700px">
+                <${tableColumnIconTag} ${ref('col1')} field-name="field1" key-type="${keyType}">
+                    Column 1
+                    ${repeat(() => mappings, html<BasicIconMapping>`
+                        <${mappingIconTag}
+                            key="${x => x.key}"
+                            text="${x => x.text}"
+                            icon="${x => x.icon}">
+                        </${mappingIconTag}>
+                    `)}
+                </${tableColumnIconTag}>
+            </${tableTag}>`, { source });
+        return {
+            ...result,
+            model: source
+        };
+    }
+
+    afterEach(async () => {
+        if (disconnect) {
+            await disconnect();
+        }
+    });
+
+    it('should export its tag', () => {
+        expect(tableColumnIconTag).toBe('nimble-table-column-icon');
+    });
+
+    it('can construct an element instance', () => {
+        expect(
+            document.createElement('nimble-table-column-icon')
+        ).toBeInstanceOf(TableColumnIcon);
+    });
+
+    for (const test of [
```

Comment:
Is there a reason not to use the existing `getSpecTypeByNamedList` pattern here?